Fix arch detection

mfalkvidd opened the pull request:
When there was no image, the script always selected arm.
Now we use uname to get the arch instead. This will work even if
there is no image.

Minor cleanup stuff:
* Remove redundant API call to quay
* Set height_diff high if the diff can't be calculated (when the
container is not running).
* Remove unnecessary sleep

mfalkvidd committed:
Fix arch detection

When there was no image, the script always selected arm.
Now we use uname to get the arch instead. This will work even if
there is no image.

Minor cleanup stuff:
* Remove redundant API call to quay
* Set height_diff high if the diff can't be calculated (when the
container is not running).
* Remove unnecessary sleep

miqueet reviewed (changes requested):
To attempt to future proof anything can you make the following change.

`QUAY_URL='https://quay.io/api/v1/repository/team-helium/miner/tag/?limit=6&page=1&onlyActiveTags=true'`

to this

`QUAY_URL='https://quay.io/api/v1/repository/team-helium/miner/tag/?limit=20&page=1&onlyActiveTags=true'`

In case they decide to support other architectures or add additional builds i'd like to attempt to future proof it.

Other than that everything looks really good. I had been meaning to clear that up, so thanks again!

mfalkvidd committed:
Get 20 items from Quay

If Helium add many new architectures or variants, 6 might not be
enough.

mfalkvidd committed:
Remove faulty height_diff

Somehow, this resulted in height_diff being set to 999 every time,
which caused the miner to be restarted every time.

mfalkvidd committed:
Remove stray semicolon

Wheaties466 commented:
Merging.
